File: xmuoj/contest-88/D1P3-problem/main.cpp
#include <iostream>![1776672456430](image/main/1776672456430.png)
using namespace std;

int main() {
    int p, e, i, d;
    int caseNum = 1;

    while (cin >> p >> e >> i >> d) {
        if (p == -1 && e == -1 && i == -1 && d == -1) {
            break;
        }

        int n = d + 1;
        bool found = false;
        while (n <= d + 21252) {
            if ((n - p) % 23 == 0 && (n - e) % 28 == 0 && (n - i) % 33 == 0 && n > d) {
                cout << "Case " << caseNum << ": the next triple peak occurs in " << (n - d) << " days." << endl;
                found = true;
                break;
            }
            n++;
        }

        if (!found) {
            for (int step = 0; step <= 21252; step++) {
                n = p + step * 23;
                if (n > d && n % 28 == e && n % 33 == i) {
                    cout << "Case " << caseNum << ": the next triple peak occurs in " << (n - d) << " days." << endl;
                    found = true;
                    break;
                }
            }
        }

        caseNum++;
    }

    return 0;
}

File: xmuoj/contest-88/D1P3-problem/main.py
def solve():
    case_num = 1
    while True:
        line = input()
        if not line:
            break
        p, e, i, d = map(int, line.split())
        if p == -1 and e == -1 and i == -1 and d == -1:
            break

        n = d + 1
        found = False
        while n <= d + 21252:
            if n % 23 == p % 23 or n % 23 == (p % 23 + 23) % 23:
                pass
            if (n - p) % 23 == 0 and (n - e) % 28 == 0 and (n - i) % 33 == 0 and n > d:
                print(f"Case {case_num}: the next triple peak occurs in {n - d} days.")
                found = True
                break
            n += 1
        if not found:
            for step in range(21253):
                n = p + step * 23
                if n > d and n % 28 == e % 28 and n % 33 == i % 33:
                    print(f"Case {case_num}: the next triple peak occurs in {n - d} days.")
                    found = True
                    break
        case_num += 1

solve()